src/cpp/rtps/xmlparser/XMLElementParser.cpp

```diff
@@ -1058,13 +1058,14 @@ XMLP_ret XMLParser::getXMLFlowControllerDescriptorList(
             {
                 std::lock_guard<std::mutex> lock(collections_mtx_);
                 // name - stringType
-                flow_controller_descriptor_names_.emplace_back(get_element_text(p_aux1));
-                flow_controller_descriptor->name = flow_controller_descriptor_names_.back().c_str();
-                if (flow_controller_descriptor_names_.back().empty())
+                auto element_inserted = flow_controller_descriptor_names_.insert(get_element_text(p_aux1));
```

Comment:
I would separate the check for the element. So call `get_element_text` and validate the returned value, then insert in the set